pyccel/codegen/printing/ccode.py

```diff
@@ -1404,18 +1405,30 @@ def _print_AugAssign(self, expr):
         return "{0} {1}= {2};\n".format(lhs_code, op, rhs_code)
 
     def _print_Assign(self, expr):
-        if isinstance(expr.rhs, FunctionCall) and isinstance(expr.rhs.dtype, NativeTuple):
-            self._temporary_args = [VariableAddress(a) for a in expr.lhs]
-            return '{};\n'.format(self._print(expr.rhs))
-        if isinstance(expr.rhs, (NumpyArray)):
-            return self.copy_NumpyArray_Data(expr)
-        if isinstance(expr.rhs, (NumpyFull)):
-            return self.arrayFill(expr)
-        if isinstance(expr.rhs, NumpyArange):
-            return self.fill_NumpyArange(expr.rhs, expr.lhs)
+        prefix_code = ''
+        lhs = expr.lhs
+        rhs = expr.rhs
+        if isinstance(lhs, Variable) and lhs.is_optional:
+            if lhs in self._optional_partners:
+                tmp_var = self._optional_partners[lhs]
+            else:
+                tmp_var_name = self._parser.get_new_name()
+                tmp_var = lhs.clone(tmp_var_name, is_optional=False)
+                self._additional_declare.append(tmp_var)
+                self._optional_partners[lhs] = tmp_var
+            prefix_code = self._print(AliasAssign(lhs, tmp_var))
```

Comment:
Could you explain with some comments in the code what `self._optional_partners` is and how to use it? This is beneficial for future developers.

Reply:
Is this sufficiently clear? 4014166

Reply:
It is indeed,